waspc/data/Cli/templates/basic/schema.prisma

```diff
@@ -0,0 +1,12 @@
+// Wasp uses the datasource you specify but overwrites the `url` field.
```

Comment:
Hmmm yeah I get what you mean. I also don't like it that they have to know about implementation details here. Tha is a con of the "error" method.

But "override" method has another con, which is that what is written in their Prisma file is not true, that is what bothers me. If they have `url = env("DATABASE_URL")` but that is not true, and instead db is in `file:./dev.db`, that is straight wrong. They might be trying to set up DATABASE_URL, and nothing will be happening. They can ask GPT for help, it will give them wrong help. Even some Prisma tooling might be working wrong, because it is believing the Prisma file. 

So the question is, what is a bigger issue of those two. But to me, the "it is not true" element of override seems like a quite bigger issue.

Reply:
> But "override" method has another con, which is that what is written in their Prisma file is not true, that is what bothers me. If they have url = env("DATABASE_URL") but that is not true, and instead db is in file:./dev.db, that is straight wrong

But even if they wrote in the Prisma file exactly what we would write ourselves: `file:./dev.db` it still would be wrong since the Prisma file they work on is not the one our tooling uses, it's rather the `.wasp/out/db/schema.prisma` file and the DB is located there `.waps/out/db/dev.db`.  So, the file cannot tell the truth either way. 

That's why I'm even more for making it broken in a sense that's it's obvious that this file is not _the schema file_ 100% without preprocessing. Maybe even go so far as to put `url = "wasp"` and then explain above "Wasp manages this field and it's replaces by the compiler" or smth like that.

Reply:
Uf yeah I forgot about the fact that SQLite is in another location! I agree, that is also not great.

Ok, I do agree that we should then go with either:
1. Making it obvious it is a not true.
2. Making it true.

We can make it obvious it is not true by possibly doing something like `env("WASP_WILL_SET_THIS")`. But that doesn't feel great.

We could make it true! So it has to be `./wasp/out/db/dev.db` for sqlite? But this is indeed kind of implementation detail, it is also weird.

Ok, so `env('DATABASE_URL')` is probably the best! But hey, idea: can we make that completely true? It is already true for all cases but for sqlite -> but we can also provide the URL to database via DATABASE_URL for it in Wasp instead of hardcoding it into generated prisma schema, and that is it then?

So, my suggestion is:
1. We enforce them to set `url` to `env('DATABASE_URL')`. If they don't, we tell them it has to be.
2. When they use sqlite, we modify generator to not hardcode URL into generated prisma schema but to also use DATABASE_URL env, and we also provide env which has `file:./dev.db`. I am not 100% sure this will work, but it should, right?

Reply:
This is what I went for:
> We enforce them to set url to env('DATABASE_URL'). If they don't, we tell them it has to be.

and

> When they use sqlite, we modify generator to not hardcode URL into generated prisma schema but to also use DATABASE_URL env

Great suggestions 👍

Reply:
Awesome! We found the best solution together :)!

You can resolve this one.